### ISSUE TITLE
stat.py reports file does not exist but it does

### ISSUE THREAD

**Issue by treblesome** [ansible/ansible#39905]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME

stat.py

##### ANSIBLE VERSION

```
[root]snaxl96/> ansible --version
ansible 2.4.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/freeware/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.13 (default, Oct 17 2017, 05:37:56) [C]
```

##### CONFIGURATION

[root]snaxl96/> ansible-config dump --only-changed
DEFAULT_HOST_LIST(/etc/ansible/ansible.cfg) = [u'/etc/ansible/hosts']

##### OS / ENVIRONMENT

AIX 7100-04-04-1717

##### SUMMARY

When copying a file to a remote server using the copy module, one remote server fails and return an error as if the file does not exist.... at least that's how I understand it by looking at the verbose output.

##### STEPS TO REPRODUCE

**Test file:**
[root]snaxl96/> ls -al /tmp/snaxl96.f1
-rw-r--r--    1 root     system           21 May  1 14:41 /tmp/snaxl96.f1

[root]snaxl96/> ansible snaxl94 -m copy -a "src=/tmp/snaxl96.f1 dest=/tmp/"
snaxl94 | FAILED! => {
    "msg": "module (copy) is missing interpreter line"
}

**Same command with -vvv option:**
...
<snaxl94> (0, '\r\n{"invocation": {"module_args": {"checksum_algorithm": "sha1", "get_checksum": true, "path": "/tmp/snaxl96.f1", "checksum_algo": "sha1", "follow": false, "get_md5": false, "get_mime": true, "get_attributes": true}}, **"stat": {"exists": false},** "changed": false}\r\n', 'Shared connection to snaxl94 closed.\r\n')

**Running the same command on another remote server:**
[root]snaxl96/> ansible snaxl95 -m copy -a "src=/tmp/snaxl96.f1 dest=/tmp/"
snaxl95 | SUCCESS => {
    "changed": false,
    "checksum": "5d3b1b2104a778ce68a788672b8cdcd8013d7540",
    "gid": 0,
    "group": "system",
    "mode": "0644",
    "owner": "root",
    "path": "/tmp/snaxl96.f1",
    "size": 21,
    "state": "file",
    "uid": 0
}

**With -vvv option showing only the stat line:**
<snaxl95> (0, '\r\n{"invocation": {"module_args": {"checksum_algorithm": "sha1", "get_checksum": true, "path": "/tmp/snaxl96.f1", "checksum_algo": "sha1", "follow": false, "get_md5": false, "get_mime": true, "get_attributes": true}}, "**stat": {"charset": "unknown"**, "uid": 0, "exists": true, "attr_flags": "", "generation": -979456329, "woth": false, "isreg": true, "device_type": 0, "mtime": 1525790207.0, "block_size": 4096, "inode": 10, "isgid": false, "size": 21, "executable": true, "isuid": false, "readable": true, "version": null, "pw_name": "root", "gid": 0, "ischr": false, "wusr": true, "writeable": true, "mimetype": "unknown", "blocks": 8, "xoth": false, "islnk": false, "nlink": 1, "issock": false, "rgrp": true, "gr_name": "system", "path": "/tmp/snaxl96.f1", "xusr": false, "atime": 1525872659.0, "isdir": false, "ctime": 1525790208.0, "isblk": false, "wgrp": false, "checksum": "5d3b1b2104a778ce68a788672b8cdcd8013d7540", "dev": 655367, "roth": true, "isfifo": false, "mode": "0644", "xgrp": false, "rusr": true, "attributes": []}, "changed": false}\r\n', 'Shared connection to snaxl95 closed.\r\n')

**Python on remote servers:**
snaxl94/>python --version
Python 2.7.13

dr-snaxl95/>python --version
Python 2.7.13

```yaml
ansible snaxl94 -m copy -a "src=/tmp/snaxl96.f1 dest=/tmp/"
```

##### EXPECTED RESULTS

That it will copy the local file to the remote host, snaxl94.

##### ACTUAL RESULTS

[root]snaxl96/> ansible -vvv snaxl94 -m copy -a "src=/tmp/snaxl96.f1 dest=/tm>
ansible 2.4.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/freeware/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.13 (default, Oct 17 2017, 05:37:56) [C]
Using /etc/ansible/ansible.cfg as config file
Parsed /etc/ansible/hosts inventory source with ini plugin
META: ran handlers
<snaxl94> ESTABLISH SSH CONNECTION FOR USER: None
<snaxl94> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/.ansible/cp/25e7a432a7 snaxl94 '/bin/sh -c '"'"'echo ~ && sleep 0'"'"''
<snaxl94> (0, '/root\n', '')
<snaxl94> ESTABLISH SSH CONNECTION FOR USER: None
<snaxl94> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/.ansible/cp/25e7a432a7 snaxl94 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751 `" && echo ansible-tmp-1525873257.63-214960765758751="` echo /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751 `" ) && sleep 0'"'"''
<snaxl94> (0, 'ansible-tmp-1525873257.63-214960765758751=/root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751\n', '')
Using module file /opt/freeware/lib/python2.7/site-packages/ansible/modules/files/stat.py
<snaxl94> PUT /tmp/tmpD6H3V7 TO /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751/stat.py
<snaxl94> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/.ansible/cp/25e7a432a7 '[snaxl94]'
<snaxl94> (0, 'sftp> put /tmp/tmpD6H3V7 /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751/stat.py\nUploading /tmp/tmpD6H3V7 to /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751/stat.py\n', '')
<snaxl94> ESTABLISH SSH CONNECTION FOR USER: None
<snaxl94> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/.ansible/cp/25e7a432a7 snaxl94 '/bin/sh -c '"'"'chmod u+x /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751/ /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751/stat.py && sleep 0'"'"''
<snaxl94> (0, '', '')
<snaxl94> ESTABLISH SSH CONNECTION FOR USER: None
<snaxl94> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/.ansible/cp/25e7a432a7 -tt snaxl94 '/bin/sh -c '"'"'/usr/bin/python /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751/stat.py && sleep 0'"'"''
<snaxl94> (0, '\r\n{"invocation": {"module_args": {"checksum_algorithm": "sha1", "get_checksum": true, "path": "/tmp/snaxl96.f1", "checksum_algo": "sha1", "follow": false, "get_md5": false, "get_mime": true, "get_attributes": true}}, "stat": {"exists": false}, "changed": false}\r\n', 'Shared connection to snaxl94 closed.\r\n')
<snaxl94> PUT /tmp/snaxl96.f1 TO /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751/source
<snaxl94> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/.ansible/cp/25e7a432a7 '[snaxl94]'
<snaxl94> (0, 'sftp> put /tmp/snaxl96.f1 /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751/source\nUploading /tmp/snaxl96.f1 to /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751/source\n', '')
<snaxl94> ESTABLISH SSH CONNECTION FOR USER: None
<snaxl94> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/.ansible/cp/25e7a432a7 snaxl94 '/bin/sh -c '"'"'chmod u+x /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751/ /root/.ansible/tmp/ansible-tmp-1525873257.63-214960765758751/source && sleep 0'"'"''
<snaxl94> (0, '', '')
Using module file /opt/freeware/lib/python2.7/site-packages/ansible/modules/files/copy.mod
snaxl94 | FAILED! => {
    "msg": "module (copy) is missing interpreter line"
}

```

```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/files/stat.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/files/stat.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @bpennypacker
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by treblesome** [comment]
Update: resolved

Apparently, the problem was not on the stat module but the copy module. So I copied copy.py from a 2.5 and ran the ad-hoc command again and this time, it worked. 

[root]snaxl96/> ansible snaxl94 -m copy -a "src=/tmp/snaxl96.f1 dest=/tmp/"
snaxl94 | SUCCESS => {
    "changed": false,
    "checksum": "5d3b1b2104a778ce68a788672b8cdcd8013d7540",
    "gid": 0,
    "group": "system",
    "mode": "0644",
    "owner": "root",
    "path": "/tmp/snaxl96.f1",
    "size": 21,
    "state": "file",
    "uid": 0
}

2.4 copy.py:
-rw-r--r--    1 root     system        15146 Mar  2 04:23 copy.py.ORIG

2.5 copy.py:
-rw-r--r--    1 root     system        15389 May  9 14:43 copy.py

There is quite a difference in copy.py between 2.4 and 2.5 but I'm not going to list it here. Just know the problem is fixed on ansible 2.5.

Closing this issue.
